collector/collector_test.go

```diff
@@ -300,6 +309,158 @@ info:
+          passwd: bar
+`), os.ModePerm)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				err = os.RemoveAll(tmpDir1)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("keeps the two users", func() {
+				o := &Options{}
+				err := o.Apply(
+					MergeBootLine,
+					WithBootCMDLineFile(cmdLinePath),
+					Directories(tmpDir1),
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				c, err := Scan(o, FilterKeysTestMerge)
+				Expect(err).ToNot(HaveOccurred())
+
+				fmt.Println(c.String())
+				Expect(c.String()).To(Equal(`#cloud-config
+
+install:
+    auto: true
+    grub_options:
+        extra_cmdline: console=tty0
+    poweroff: false
+    reboot: false
+options:
+    device: /dev/sda
+stages:
+    initramfs:
+        - users:
+            kairos:
+                groups:
+                    - sudo
+                passwd: kairos
+        - users:
+            foo:
+                groups:
+                    - sudo
+                passwd: bar
+`))
+			})
+		})
+
+		Context("When a YIP if expression is contained", func() {
+			var cmdLinePath, tmpDir1 string
+			var err error
+
+			BeforeEach(func() {
+				tmpDir1, err = os.MkdirTemp("", "config1")
+				Expect(err).ToNot(HaveOccurred())
+				err := os.WriteFile(path.Join(tmpDir1, "local_config_1.yaml"), []byte(`#cloud-config
+stages:
+  initramfs:
+  - users:
+      kairos:
+        passwd: kairos
+  - name: set_inotify_max_values
+    if: '[ ! -f /oem/80_stylus.yaml ]'
+    sysctl:
+      fs.inotify.max_user_instances: "8192"
+`), os.ModePerm)
+				Expect(err).ToNot(HaveOccurred())
+				err = os.WriteFile(path.Join(tmpDir1, "local_config_2.yaml"), []byte(`#cloud-config
+stages:
+  initramfs:
+  - commands:
+    - ln -s /etc/kubernetes/admin.conf /run/kubeconfig
+    sysctl:
+      kernel.panic: "10"
+      kernel.panic_on_oops: "1"
+      vm.overcommit_memory: "1"
+`), os.ModePerm)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				err = os.RemoveAll(tmpDir1)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("it remains within its scope after merging", func() {
+				o := &Options{}
+				err := o.Apply(
+					MergeBootLine,
+					WithBootCMDLineFile(cmdLinePath),
+					Directories(tmpDir1),
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				c, err := Scan(o, FilterKeysTestMerge)
+				Expect(err).ToNot(HaveOccurred())
+
+				fmt.Println(c.String())
+				Expect(c.String()).To(Equal(`#cloud-config
+
+stages:
+    initramfs:
+        - users:
+            kairos:
+                passwd: kairos
+        - if: '[ ! -f /oem/80_stylus.yaml ]'
+          name: set_inotify_max_values
+          sysctl:
+            fs.inotify.max_user_instances: "8192"
+        - commands:
+            - ln -s /etc/kubernetes/admin.conf /run/kubeconfig
+          sysctl:
+            kernel.panic: "10"
+            kernel.panic_on_oops: "1"
+            vm.overcommit_memory: "1"
+`))
+			})
+		})
+
 		Context("duplicated configs", func() {
```

Comment:
```suggestion
		Context("duplicated slice configs", func() {
```

I guess duplicated "maps" will still be merged right?

Reply:
yeah this one includes slices and maps, e.g. initramfs in this case